api/handlers_metrics.go

```diff
@@ -206,3 +206,81 @@ func parseQuery(r *http.Request, request cloudwatch.MetricsRequest) error {
 
 	return nil
 }
+
+// GetRDSMetricsURLHandler gets RDS metrics from cloudwatch and returns a link to the image
+func (s *server) GetRDSMetricsURLHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	instanceId := vars["id"]
+
+	cwService, ok := s.cloudwatchServices[account]
+	if !ok {
+		msg := fmt.Sprintf("cloudwatch service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+	log.Debugf("found cloudwatch service %+v", cwService)
+
+	resultCache, ok := s.resultCache[account]
+	if !ok {
+		msg := fmt.Sprintf("result cache not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+	log.Debugf("found cost explorer result cache %+v", *resultCache)
+
+	queries := r.URL.Query()
+	metrics := queries["metric"]
+	if len(metrics) == 0 {
+		handleError(w, apierror.New(apierror.ErrBadRequest, "at least one metric is required", nil))
+		return
+	}
+
+	req := cloudwatch.MetricsRequest{}
+	if err := parseQuery(r, req); err != nil {
+		handleError(w, apierror.New(apierror.ErrBadRequest, "failed to parse query", err))
+		return
+	}
+
+	key := fmt.Sprintf("%s/%s/%s/%v/%v/%v", Org, instanceId, strings.Join(metrics, "-"), req["start"], req["end"], req["period"])
+	hashedCacheKey := s.imageCache.HashedKey(key)
+	if res, expire, ok := resultCache.GetWithExpiration(hashedCacheKey); ok {
+		log.Debugf("found cached object: %s", res)
+
+		if body, ok := res.([]byte); ok {
+			w.Header().Set("X-Cache-Hit", "true")
+			w.Header().Set("X-Cache-Expire", fmt.Sprintf("%0.fs", time.Until(expire).Seconds()))
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			w.Write(body)
+			return
+		}
+	}
+
+	cwMetrics := []cloudwatch.Metric{}
+	for _, m := range metrics {
+		cwMetrics = append(cwMetrics, cloudwatch.Metric{"AWS/RDS", m, "DBInstanceIdentifier", instanceId})
```

Comment:
I know we talked about this but I can't remember 😖 Do we need to think about the cluster metrics too?  I think those are for aurora?

Reply:
I added the DBClusterIdentifier for aurora